src/useFilterMessages.ts

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+import type {
+  IDecodedMessage,
+  IDecoder,
+  IFilter,
+  Waku,
+} from "@waku/interfaces";
+
+import type { HookState } from "./types";
+
+type AbstractFilterNode = Waku & {
+  filter: IFilter;
+};
+
+type UseFilterMessagesParams = {
+  node: undefined | AbstractFilterNode;
+  decoder: undefined | IDecoder<IDecodedMessage>;
```

Comment:
```suggestion
  decoder: undefined | IDecoder<T extends IDecodedMessage>;
```

Not sure how much it matter but the developer will not use `IDecodedMessage` but a class that implements it.

Reply:
It is a good clarification, but from the types point does it make difference? We care only how it will be used and `IDecodedMessage` gives enough info. Making it generic will make usage of the hook a bit harder.